### PR TITLE
bank_hash_at uses rewrites

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Summary of Changes

Fixes #

### CODE REVIEW

File: runtime/src/accounts_db.rs [hunk 1]
```diff
@@ -5992,6 +5993,9 @@ impl AccountsDb {
     }
 
     pub fn get_accounts_delta_hash(&self, slot: Slot) -> Hash {
+        self.get_accounts_delta_hash_new(slot, &Rewrites::default())
+    }
+    pub fn get_accounts_delta_hash_new(&self, slot: Slot, rewrites: &Rewrites) -> Hash {
```
Comment (brooksprumo) [inline runtime/src/accounts_db.rs:5998]: nit: What about `get_accounts_delta_hash_with_rewrites` instead?

File: runtime/src/accounts_db.rs [hunk 2]
```diff
@@ -6054,6 +6060,15 @@ impl AccountsDb {
         ret
     }
 
+    /// add all items from rewrites to hashes if the pubkey doesn't already exist in hashes
+    fn extend_hashes_with_rewrites(hashes: &mut Vec<(Pubkey, Hash)>, rewrites: &Rewrites) {
+        let mut rewrites = rewrites.read().unwrap().clone();
```
[the remark this answers is not in the document]
Comment (jeffwashington) [reply runtime/src/accounts_db.rs:6065]: We will skip every rewrite. This means we will skip far more than we will write per slot. I just spent a few minutes re-evaluating this algorithm. We need the ability to lookup the skipped rewrites, so hashmap is the right structure. The pubkeys in the storage are returned as a dashmap, I think. The pubkeys in the write cache are returned as a vec. So, we need to merge these two 'lists' of (pubkey, hash). If the storage or write cache has an entry for a pubkey, then that one takes precedence over the skipped rewrite. I don't see an obvious way to improve on what is here. I need to linearly search the storage entries/cache entries and eliminate the duplicates from the hashmap.

File: runtime/src/accounts_db.rs (hunk 1 above)
Comment (jeffwashington) [reply runtime/src/accounts_db.rs]: renamed

File: runtime/src/accounts_db.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (jeffwashington) [reply runtime/src/accounts_db.rs]: I renamed some things and added a test.

[the remark this answers is not in the document]
Comment (brooksprumo) [reply runtime/src/accounts_db.rs]: Yeah, I can't think of any impl/algo that's obviously better. I'm guessing that as this gets some runtime we'll see if it's obviously slow and needs to be tweaked.

**Review by brooksprumo — APPROVED** [review]
lgtm

**Pull request merged**